### PR TITLE
Ag 8576/waterfall series animations

### PR DESCRIPTION
**Pull request by manapeirov**
PR for https://ag-grid.atlassian.net/browse/AG-8576

### CODE REVIEW

**Review by alexanderby — COMMENTED** [review]
Don't give up.

File: packages/ag-charts-enterprise/src/waterfall/waterfallSeries.ts [hunk 1]
```diff
@@ -781,15 +782,14 @@ export class WaterfallBarSeries extends _ModuleSupport.CartesianSeries<
...
-        seriesRect,
     }: {
         datumSelections: Array<_Scene.Selection<_Scene.Rect, WaterfallNodeDatum>>;
         labelSelections: Array<_Scene.Selection<_Scene.Text, WaterfallNodeDatum>>;
         contextData: Array<WaterfallContext>;
         paths: Array<Array<_Scene.Path>>;
         seriesRect?: _Scene.BBox;
     }) {
-        const duration = this.ctx.animationManager?.defaultOptions.duration ?? 1000;
+        const duration = this.ctx.animationManager?.defaultOptions.duration ?? 2000;
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/waterfall/waterfallSeries.ts]: 1000

File: packages/ag-charts-enterprise/src/waterfall/waterfallSeries.ts [hunk 2]
```diff
@@ -829,56 +830,52 @@ export class WaterfallBarSeries extends _ModuleSupport.CartesianSeries<
...
-        lineNode.lineJoin = 'round';
-        lineNode.pointerEvents = _Scene.PointerEvents.None;
+        const valueAxis = this.getValueAxis();
+        const startX = valueAxis?.scale.convert(0);
+        const endX = pointData.reduce((end, point) => {
+            if (point.x > end) {
+                end = point.x;
+            }
+            return end;
+        }, 0);
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/waterfall/waterfallSeries.ts]: `-Infinity`? Can points be negative?

File: packages/ag-charts-enterprise/src/waterfall/waterfallSeries.ts [hunk 3]
```diff
@@ -800,22 +800,23 @@ export class WaterfallBarSeries extends _ModuleSupport.CartesianSeries<
...
+        datumSelection.each((rect, datum) => {
             this.ctx.animationManager?.animateMany(
                 `${this.id}_empty-update-ready_${rect.id}`,
                 [
-                    { from: datum.itemId === 'positive' ? datum.x : datum.x + datum.width, to: datum.x },
+                    { from: yAxis?.scale.convert(0) ?? 0, to: datum.x },
                     { from: 0, to: datum.width },
                 ],
                 {
                     disableInteractions: true,
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/waterfall/waterfallSeries.ts]: This defaults to `true` now, so can be removed

File: packages/ag-charts-enterprise/src/waterfall/waterfallSeries.ts [hunk 4]
```diff
@@ -1001,18 +1002,59 @@ export class WaterfallColumnSeries extends WaterfallBarSeries {
...
+        datumSelection.each((rect, datum) => {
             this.ctx.animationManager?.animateMany(
                 `${this.id}_empty-update-ready_${rect.id}`,
                 [
-                    { from: datum.itemId === 'positive' ? datum.y + datum.height : datum.y, to: datum.y },
+                    { from: yAxis?.scale.convert(0) ?? 0, to: datum.y },
                     { from: 0, to: datum.height },
                 ],
                 {
                     disableInteractions: true,
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/waterfall/waterfallSeries.ts]: and here

**Review by lsjroberts — APPROVED** [review]

**Review by lsjroberts — APPROVED** [review]

**Pull request merged**